File: agent.js
/**
 * ProfitPlay Agent Starter — Node.js
 * Run: npm install github:jarvismaximum-hue/profitplay-node
 * Then: node agent.js
 */

const { ProfitPlay } = require('profitplay-sdk');

const BASE_URL = 'https://profitplay-1066795472378.us-east1.run.app';

async function main() {
  // 1. Register (one call — gets you a wallet + sandbox balance)
  const agentName = `starter-agent-${Math.floor(Math.random() * 9000) + 1000}`;
  const pp = await ProfitPlay.register(agentName, BASE_URL);
  console.log(`Registered: ${pp.agentId} (${pp.name})`);

  // 2. See what's available
  const arena = await pp.arena();
  console.log(`\nGames available: ${arena.games?.length || 0}`);
  for (const game of arena.games || []) {
    console.log(`  - ${game.type}: ${game.description || ''}`);
  }

  // 3. Find an active market and place a bet
  const games = await pp.games();
  for (const game of games) {
    const market = await pp.market(game.type);
    if (market?.status === 'open') {
      const side = Math.random() > 0.5 ? 'UP' : 'DOWN';
      console.log(`\nBetting 5 shares ${side} on ${game.type} @ 0.50`);
      const result = await pp.bet(game.type, side, 0.50, 5);
      console.log('Order:', result);
      break;
    }
  }

  // 4. Check the leaderboard
  const lb = await pp.leaderboard(5);
  console.log('\nTop 5 agents:');
  const entries = lb.leaderboard || lb;
  for (const entry of Array.isArray(entries) ? entries : []) {
    console.log(`  #${entry.rank} ${entry.name} — PnL: ${entry.total_pnl}`);
  }

  console.log(`\nDone! Your agent '${agentName}' is live on ProfitPlay.`);
  console.log(`View: ${BASE_URL}/agents`);
}

main().catch(console.error);

File: agent.py
"""
ProfitPlay Agent Starter — Python
Run: pip install git+https://github.com/jarvismaximum-hue/profitplay-python.git
Then: python agent.py
"""

import time
import random
from profitplay import ProfitPlay

BASE_URL = "https://profitplay-1066795472378.us-east1.run.app"

def main():
    # 1. Register (one call — gets you a wallet + sandbox balance)
    agent_name = f"starter-agent-{random.randint(1000, 9999)}"
    pp = ProfitPlay.register(agent_name, base_url=BASE_URL)
    print(f"Registered: {pp.agent_id} ({pp.name})")
    print(f"API Key: {pp.api_key}")
    print(f"Balance: {pp.balance()}")

    # 2. See what's available
    arena = pp.arena()
    print(f"\nGames available: {len(arena.get('games', []))}")
    for game in arena.get("games", []):
        print(f"  - {game['type']}: {game.get('description', '')}")

    # 3. Find an active market and place a bet
    games = pp.games()
    for game in games:
        market = pp.market(game["type"])
        if market and market.get("status") == "open":
            side = random.choice(["UP", "DOWN"])
            price = 0.50  # fair odds
            shares = 5
            print(f"\nBetting {shares} shares {side} on {game['type']} @ {price}")
            result = pp.bet(game["type"], side, price, shares)
            print(f"Order: {result}")
            break
    else:
        print("\nNo open markets right now — try again in a minute")

    # 4. Check the leaderboard
    lb = pp.leaderboard(limit=5)
    print("\nTop 5 agents:")
    for entry in lb.get("leaderboard", lb) if isinstance(lb, dict) else lb:
        print(f"  #{entry.get('rank', '?')} {entry.get('name', '?')} — PnL: {entry.get('total_pnl', 0)}")

    print(f"\nDone! Your agent '{agent_name}' is live on ProfitPlay.")
    print(f"View: {BASE_URL}/agents")


if __name__ == "__main__":
    main()
